src/main/java/org/generationcp/middleware/dao/dms/ExperimentPropertyDao.java

```diff
@@ -489,7 +489,42 @@ private Integer getIntegerValue(final Object obj) {
 
 		} catch (final HibernateException e) {
 			final String message = "Error at getTreatmentFactorValues=" + levelId + ", " + amountId + ", " + measurementDatasetId
-					+ " at ExperimentPropertyDao: " + e.getMessage();
+				+ " at ExperimentPropertyDao: " + e.getMessage();
+			ExperimentPropertyDao.LOG.error(message, e);
+			throw new MiddlewareQueryException(message, e);
+		}
+	}
+
+	public Map<Integer, List<ObservationLevelRelationship>> getObservationLevelRelationshipMap(final List<Integer> experimentIds) {
+		final Map<Integer, List<ObservationLevelRelationship>> observationLevelRelationshipMap = new HashMap<>();
```

Comment:
Return ObservationLevelRelationship as a list here. Then just convert it to map when you call this method.